strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java

```diff
@@ -402,27 +429,61 @@ public void testRemoveTimestampedSnapshots(@MavenRepository(repositoryId = TRTS_
 
+        // To check removing timestamped snapshot with keepPeriod = 3 and numberToKeep = 0
+        mavenRepositoryFeatures.removeTimestampedSnapshots(storageId,
+                                                           repositoryId,
+                                                           "org/carlspring/strongbox/timestamped",
+                                                           0,
+                                                           3);
+
+        // Check all remaining artifacts from the repository.
+        Path repositoryPath = repositoryPathResolver.resolve(repository);
+        try (Stream<Path> pathStream = Files.walk(repositoryPath))
+        {
+            long timestampedSnapshots = pathStream.filter(path -> path.toString().endsWith(".jar")).count();
+            assertEquals(1, timestampedSnapshots, "Amount of timestamped snapshots doesn't equal 1.");
+
+            Path snapshotArtifactPath = getSnapshotArtifactPath(repositoryPath);
+            assertNotNull(snapshotArtifactPath);
+            assertTrue(snapshotArtifactPath.toString().endsWith("-3.jar"));
         }
+    }
 
-        //Creating timestamped snapshot with another timestamp
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    @Test
+    public void testRemoveTimestampedSnapshotsWithPreviousDate(@MavenRepository(repositoryId = TRTS_SNAPSHOTS,
+                                                                                policy = RepositoryPolicyEnum.SNAPSHOT)
+                                                               Repository repository,
+                                                               @MavenTestArtifact(repositoryId = TRTS_SNAPSHOTS,
+                                                                                  id = "org.carlspring.strongbox:timestamped",
+                                                                                  versions = { "2.0-20190626.190020-1",
+                                                                                               "2.0-20190626.190145-2"})
+                                                               List<Path> artifactPaths)
+            throws Exception
+    {
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
+
+        // Generate timestamp for current date minus 5 days
+        String timestamp = getTimestamp(5);
 
-        Calendar cal = Calendar.getInstance();
-        cal.add(Calendar.DATE, -5);
-        String timestamp = new SimpleDateFormat("yyyyMMdd.HHmmss").format(cal.getTime());
+        // Set artifact timestamp with current date minus 5 days (this can't be done with @MavenTestArtifact)
+        // Redeploy artifact with new timestamp, necessary for removeTimestampedSnapshots and keepPeriod attribute.
+        MavenArtifact artifact = setArtifactsTimestampAndRedeploy(artifactPaths,
```

Comment:
@ptirador can you please create a follow up task to improve this test and drop the `setArtifactsTimestampAndRedeploy` usage.

Reply:
#1443